include/alpaka/core/CallbackThread.hpp

```diff
@@ -54,30 +53,35 @@ namespace alpaka::core
 
         auto submit(Task task) -> std::future<void>
         {
-            auto f = task.get_future();
+            auto tp = std::make_pair(std::move(task), std::promise<void>{});
```

Comment:
Please add a comment here that we do NOT use the `packaged_task`'s future, because it keeps the task alive.